lib/logstash/codecs/netflow.rb

```diff
@@ -343,7 +343,13 @@ def decode_ipfix(flowset, record)
           when /^flow(?:Start|End)(Milli|Micro|Nano)seconds$/
             case $1
             when 'Milli'
-              event[@target][k.to_s] = LogStash::Timestamp.at(v.snapshot.to_f / 1_000).to_iso8601
+              secs = v.snapshot.to_i / 1000
+              micros = (v.snapshot.to_i % 1000) * 1000
+              # Use the 2 args Timestamp.at to avoid the precision under milliseconds. Doing math division (like /1000 on a float)
+              # could introduce error in representation that makes 0.192 millis to be expressed like 0.192000001 nanoseconds,
+              # so here we cut to millis, but there is a rounding when representing to to_iso8601, so 191998 micros becomes
+              # 192 millis in LogStash 8 while in previous versions it appears truncated like 191.
+              event[@target][k.to_s] = LogStash::Timestamp.at(secs, micros).to_iso8601
```

Comment:
Note for reviewers:
This force to update the test for `flow[Start|End]Milliseconds`, while in previous case when using `LogStash::Timestamp.at(milliseconds)` builder method resulted in cutting the to the 3 least significant digits, with this change is applies a rounding.
So the `191998` microseconds are rounded to `192`9 while previously was "cutted" at `191`. This is the reason why some test were updated.

The motivation for this change is that with the previous code, in LS8 Timestamp that resulted in a value with nanoseconds precision, while in LS7 was with milliseconds precision.